syntax = "proto3";

package greeting;

service CommandService
{
  rpc CreateUser(RegisterRequest) returns (UserResponse) {}
  rpc UpdateUser(UserRequest) returns (UserResponse) {}
  rpc DeleteUser(DeleteUserRequest) returns (UserResponse) {}
  rpc UpdateHighLow(ModifyLowHighRequest) returns (UserResponse) {}
}

service QueryService
{
  rpc Ping(PingMessage) returns (PingMessage) {}
  rpc LoginUser(LoginRequest) returns (UserResponse) {}
  rpc GetTicker(GetTickerRequest) returns (UserResponse) {}
  rpc GetAvaragePriceOfXDays(GetAvarageXDaysRequest) returns (UserResponse){}
  rpc GetTresholds(ThresholdsRequest) returns (UserResponse) {}
}

message PingMessage
{
  string message = 1;
}

message LoginRequest
{
  string email = 1;
  string password = 2;
}

message RegisterRequest
{
  string email = 1;
  string password = 2;
  string ticker = 3;
  string requestID = 4;
  string lowValue = 5;
  string highValue = 6;
}

message UserRequest
{
  string email = 1;
  string ticker = 2;
  string requestID = 3;
}

message DeleteUserRequest
{
  string email = 1;
}

message UserResponse
{
  string message = 1;
}

message GetTickerRequest
{
    string email = 1;
}

message GetAvarageXDaysRequest
{
  string days = 1;
  string email = 2;
}

message ModifyLowHighRequest
{
  string email = 1;
  string requestID = 2;
  string lowValue = 3;
  string highValue = 4;
}

message ThresholdsRequest
{
  string email = 1;
}